File: March 2024 GFG SOLUTION/March-07.java
//{ Driver Code Starts
//Initial Template for Java

import java.io.*;
import java.util.*;

class GFG {
    public static void main(String args[]) throws IOException {
        BufferedReader read =
            new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(read.readLine());
        while (t-- > 0) {
            int N = Integer.parseInt(read.readLine());
            String S = read.readLine();

            Solution ob = new Solution();
            System.out.println(ob.longestSubstring(S, N));
        }
    }
}

// } Driver Code Ends


//User function Template for Java
class Solution {
    static String longestSubstring(String s, int n) {
        int maxLen = 0;
        String ans = "-1";
        int i = 0, j = 0;

        while (i < n && j < n) {
            String subString = s.substring(i, j + 1);

            if (s.indexOf(subString, j + 1) != -1) {
                int len = subString.length();
                if (len > maxLen) {
                    maxLen = len;
                    ans = subString;
                }
            } else {
                i++;
            }
            j++;
        }
        return ans;
    }

};

File: March 2024 GFG SOLUTION/March-07.py
#User function Template for python3

class Solution:
    def longestSubstring(self, s, n):
        max_len = 0
        ans = "-1"
        i = j = 0

        while i < n and j < n:
            sub_string = s[i:j + 1]

            if s.find(sub_string, j + 1) != -1:
                length = len(sub_string)
                if length > max_len:
                    max_len = length
                    ans = sub_string
            else:
                i += 1
            j += 1
        
        return ans

 # Driver Code Starts
#Initial Template for Python 3

if __name__ == '__main__': 
    t = int (input ())
    for _ in range (t):
        N=int(input())
        S=input()
        
        ob = Solution()
        print(ob.longestSubstring(S , N))
# } Driver Code Ends
